eel/eel-background.c

```diff
@@ -1001,13 +1001,19 @@ void
 eel_bg_load_from_gsettings (EelBackground *self,
 			    GSettings     *settings)
 {
-    char *keyfile = g_settings_get_string (settings, MATE_BG_KEY_PICTURE_FILENAME);
+    char *picture_filename;
 
-    if (!g_file_test (keyfile, G_FILE_TEST_EXISTS) && (*keyfile != '\0'))
+    picture_filename = g_settings_get_string (settings,
+                                              MATE_BG_KEY_PICTURE_FILENAME);
+    if ((*picture_filename != '\0') &&
```

Comment:
Are you sure with the evaluation order here? I dont know if `g_settings_get_string` can return NULL?

Reply:
`g_settings_get_string` doesn't return NULL, but it can return a newly-allocated string that can be NULL
```C
/* Convenience API (get, set_string, int, double, boolean, strv) {{{1 */
/**
 * g_settings_get_string:
 * @settings: a #GSettings object
 * @key: the key to get the value for
 *
 * Gets the value that is stored at @key in @settings.
 *
 * A convenience variant of g_settings_get() for strings.
 *
 * It is a programmer error to give a @key that isn't specified as
 * having a string type in the schema for @settings.
 *
 * Returns: (not nullable) (transfer full): a newly-allocated string
 *
 * Since: 2.26
 */
gchar *
g_settings_get_string (GSettings   *settings,
                       const gchar *key)
{
  GVariant *value;
  gchar *result;

  value = g_settings_get_value (settings, key);
  result = g_variant_dup_string (value, NULL);
  g_variant_unref (value);

  return result;
}
```

Reply:
ok :)